<a href="https://colab.research.google.com/github/disha2101/1Stop_Internship_Codes/blob/main/Parkinson's_Disease_Detection_with_XGBoost.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#*Detecting Parkinson's Disease*

Author: ***Disha Bahal***

In [5]:
pip install xgboost

In [6]:
#importing necessary libraries
import numpy as np
import pandas as pd
import os,sys
from sklearn.preprocessing import MinMaxScaler
from xgboost import XGBClassifier
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score


In [7]:
#reading the dataset
df = pd.read_csv('https://raw.githubusercontent.com/chaitanyabaranwal/ParkinsonAnalysis/master/parkinsons.csv')
df.head()

name  MDVP:Fo(Hz)  MDVP:Fhi(Hz)  ...        D2       PPE  status
0  phon_R01_S01_1      119.992       157.302  ...  2.301442  0.284654       1
1  phon_R01_S01_2      122.400       148.650  ...  2.486855  0.368674       1
2  phon_R01_S01_3      116.682       131.111  ...  2.342259  0.332634       1
3  phon_R01_S01_4      116.676       137.871  ...  2.405554  0.368975       1
4  phon_R01_S01_5      116.014       141.781  ...  2.332180  0.410335       1

[5 rows x 24 columns]

In [8]:
# get the features and labels
features = df.loc[:,df.columns!='status'].values[:,1:]
labels = df.loc[:,'status'].values

In [9]:
#get the count of each label i.e 0 and 1 in labels
print(labels[labels==1].shape[0],labels[labels==0].shape[0])

147 48


In [10]:
#scale the features to between -1 and 1
scaler = MinMaxScaler((-1,1))
x=scaler.fit_transform(features)
y = labels

In [11]:
#splitting the dataset
x_train,x_test,y_train,y_test = train_test_split(x,y,test_size=0.2,random_state=7)

In [12]:
#training the model
model = XGBClassifier()
model.fit(x_train,y_train)

XGBClassifier(base_score=0.5, booster='gbtree', colsample_bylevel=1,
              colsample_bynode=1, colsample_bytree=1, gamma=0,
              learning_rate=0.1, max_delta_step=0, max_depth=3,
              min_child_weight=1, missing=None, n_estimators=100, n_jobs=1,
              nthread=None, objective='binary:logistic', random_state=0,
              reg_alpha=0, reg_lambda=1, scale_pos_weight=1, seed=None,
              silent=None, subsample=1, verbosity=1)

In [13]:
#calculating the accuracy
y_pred = model.predict(x_test)
print(accuracy_score(y_test,y_pred)*100)

94.87179487179486


*So we got 94.87% accuracy by training the model and detecting the parkinson's disease*